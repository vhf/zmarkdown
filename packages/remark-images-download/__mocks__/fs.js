const fs = jest.genMockFromModule('fs')

fs.stat = (path, callback) => {
  const err = null
  const stats = {
    isDirectory () {
      return true
    }
  }
  process.nextTick(() => callback(err, stats))
}

fs.mkdir = (path, callback) => {
  process.nextTick(() => callback())
}

module.exports = fs
