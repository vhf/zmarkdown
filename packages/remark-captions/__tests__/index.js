import {readdirSync as directory, readFileSync as file} from 'fs'
import {join} from 'path'
import ava from 'ava'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

const base = join(__dirname, 'fixtures')
const specs = directory(base).reduce((tests, contents) => {
  const parts = contents.split('.')
  if (!tests[parts[0]]) {
    tests[parts[0]] = {}
  }
  tests[parts[0]][parts[1]] = file(join(base, contents), 'utf-8')
  return tests
}, {})

const entrypoints = [
  // '../dist',
  '../src',
]

entrypoints.forEach(entrypoint => {
  const plugin = require(entrypoint)

  Object.keys(specs).filter(Boolean).forEach(name => {
    const spec = specs[name]

    ava(name, t => {
      const {contents} = unified()
        .use(reParse, {
          gfm: true,
          commonmark: false,
          yaml: false,
          footnotes: true,
          /* sets list of known blocks to nothing, otherwise <h3>hey</h3> would become
          &#x3C;h3>hey&#x3C;/h3> instead of <p>&#x3C;h3>hey&#x3C;/h3></p> */
          blocks: [],
        })
        .use(plugin)
        .use(remark2rehype)
        .use(stringify)
        .processSync(spec.fixture.replace(/Â·/g, ' '))

      t.deepEqual(contents, spec.expected.trim())
    })
  })
})
