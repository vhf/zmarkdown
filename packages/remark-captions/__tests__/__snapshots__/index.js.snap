// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code 1`] = `
"<h2>Code</h2>
<p>Normal code</p>
<pre><code class=\\"language-python\\">print('bla')
</code></pre>
<p>With Legend</p>
<figure><pre><code class=\\"language-python\\">print('bla')
</code></pre><figcaption>figcapt1</figcaption></figure>
<figure><pre><code class=\\"language-python\\">print('bla')
</code></pre><figcaption>figcapt1</figcaption></figure>
<p>break</p>
<figure><pre><code class=\\"language-python\\">print('bla')
</code></pre><figcaption>figcapt1 <em>em</em> <strong>strong <code>code</code></strong> end</figcaption></figure>
<p>bla</p>
<figure><pre><code class=\\"language-python\\">print('bla2')
</code></pre><figcaption>figcapt1</figcaption></figure>
<p>Code: bis</p>"
`;

exports[`custom-table 1`] = `
"<h2>Table</h2>
<p>Normal table</p>
<table>
<thead>
<tr>
<th>head1</th>
<th>head2</th>
</tr>
</thead>
<tbody>
<tr>
<td>bla</td>
<td>bla</td>
</tr>
</tbody>
</table>
<p>With Legend</p>
<figure><table>
<thead>
<tr>
<th>head1</th>
<th>head2</th>
</tr>
</thead>
<tbody>
<tr>
<td>bla</td>
<td>bla</td>
</tr>
</tbody>
</table><figcaption>figcapt1</figcaption></figure>
<figure><table>
<thead>
<tr>
<th>head1</th>
<th>head2</th>
</tr>
</thead>
<tbody>
<tr>
<td>bla</td>
<td>bla</td>
</tr>
</tbody>
</table><figcaption>figcapt1</figcaption></figure>
<p>CustomTable: bis</p>"
`;

exports[`external legend: two legends 1`] = `
"<figure><table>
<thead>
<tr>
<th>head1</th>
<th>head2</th>
</tr>
</thead>
<tbody>
<tr>
<td>bla</td>
<td>bla</td>
</tr>
</tbody>
</table><figcaption>figcapt1</figcaption></figure>
<p>Table: bis</p>"
`;

exports[`internal legend: two legends 1`] = `
"<p>Should only keep the 1st</p>
<figure><blockquote>
<p>My citation
Source: first capt</p>
</blockquote><figcaption>last capt<br>
2nd line</figcaption></figure>
<p>noop</p>
<figure><blockquote>
<p>foo
bar
baz
qux
Source: <strong>first</strong>
b<em>a</em>r</p>
</blockquote><figcaption>This is <strong>the real</strong> <code>source</code></figcaption></figure>
<p>noop</p>"
`;

exports[`quotation 1`] = `
"<p>(for convenience,   are replaced with
simple single spaces in the tests)</p>
<h2>Blockquote</h2>
<p>Empty blockquote</p>
<blockquote>
</blockquote>
<p>Normal blockquote</p>
<blockquote>
<p>My citation</p>
</blockquote>
<p>With Legend</p>
<figure><blockquote>
<p>My citation</p>
</blockquote><figcaption>figcapt1</figcaption></figure>
<p>With Legend without Source:</p>
<blockquote>
<p>My citation
: figcapt2</p>
</blockquote>
<p>haha</p>
<figure><blockquote>
<figure><blockquote>
<figure><blockquote>
<p>Foo<br>
Foo</p>
</blockquote><figcaption>fc1</figcaption></figure>
<p>Bar<br>
Bar<br>
</p>
</blockquote><figcaption>fc2</figcaption></figure>
<p>Baz
Baz</p>
</blockquote><figcaption>fc3</figcaption></figure>"
`;

exports[`table 1`] = `
"<h2>Table</h2>
<p>Normal table</p>
<table>
<thead>
<tr>
<th>head1</th>
<th>head2</th>
</tr>
</thead>
<tbody>
<tr>
<td>bla</td>
<td>bla</td>
</tr>
</tbody>
</table>
<p>With Legend</p>
<figure><table>
<thead>
<tr>
<th>head1</th>
<th>head2</th>
</tr>
</thead>
<tbody>
<tr>
<td>bla</td>
<td>bla</td>
</tr>
</tbody>
</table><figcaption>figcapt1</figcaption></figure>"
`;
